### ISSUE TITLE
Nan Loss after 2 epochs

### ISSUE THREAD

**Issue by harveenchadha** [SeanNaren/deepspeech.pytorch#567]
Hi,

I am receiving nan loss after 2 epochs. Can anyone guide me what can be a potential problem here?:

Things I have tried out:
1. Lowering learning rate by 100x
2. Use spec augment

<img width="683" alt="Screenshot 2020-08-18 at 12 01 00 AM" src="https://user-images.githubusercontent.com/30959215/90431278-6633eb80-e0e6-11ea-9c93-37a64bd483f1.png">

**Comment by rbracco** [comment]
This can be a number of things but the most common is that there are less frames in your input sequence than in your label. Example is a 100ms audio taken in 25ms frames with no overlap. The model will predict 4 characters, one for each frame. If the label is "house" then the probability of the correct label is 0, and the log of 0 is undefined which causes infinite loss. 

Check your input lengths and your label lengths and share more code if you are still unable to fix it.

**Comment by kouohhashi** [comment]
What kind of dataset are you using?
I experienced same thing with AN4 datasets with SGD but not with AdamW.
Or with my own datasets, junk datasets which contains wrong label caused nan loss.

**Comment by harveenchadha** [comment]
@rbracco @kouohhashi I am creating dataset by using output of Google STT as an input to model.

I have already applied the checks for input and output lengths.

I am not sure what can be wrong here. Do you advice here to check the transcripts of the audios manually? Because the data output from STT can be wrong sometimes.

**Comment by rbracco** [comment]
@harveenchadha is there a way you can share some of the code so we can look for any possible problems? Are you getting this on every item/batch? Or just occasionally? One potential issue is if you are downsampling (strided convolutions or pooling) over the time dimension too much (if I have more than a single stride 2 conv in the time dimension, I start to have these errors). 

I would also suggest manually listening to the audios and checking the transcripts. Maybe 20 or so just to validate everything looks okay. If you are having it on every batch, it probably isn't the transcripts.
